File: Php/core/FormFields/CurrencyLayoutFieldModel.php
<?php
namespace core\FormFields{
	use core\Globalization;

	class CurrencyLayoutFieldModel extends SelectFieldModel{

		public function __construct($fieldName, $fieldId, $selectedValue, $firstItemText = null){
			$items = $this->getSelectItems();
			parent::__construct($fieldName, $fieldId, $selectedValue, $items, $firstItemText);
		}

		private function getSelectItems(){
			$layouts = Globalization::getCurrencyLayouts();
			$items = array();
			if($layouts != null){
				foreach($layouts as $layout){
					$items[] = new SelectFieldItem($layout);
				}
			}
			return $items;
		}

	}

}

File: Web/Models/FormModels/CurrencyLayoutFieldModel.cs
using System.Collections.Generic;
using System.Linq;
using Infrastructure.System;

namespace Web.Models.FormModels{
    public class CurrencyLayoutFieldModel : SelectFieldModel{

		public CurrencyLayoutFieldModel(string fieldName, string fieldId, string selectedValue, string firstItemText = null)
            : base(fieldName, fieldId, selectedValue, GetSelectItems(), firstItemText)
        {
		}

		private static IEnumerable<SelectFieldItem> GetSelectItems(){
			var layouts = Globalization.GetCurrencyLayouts();
			var items = new List<SelectFieldItem>();
			if(layouts != null){
			    items.AddRange(layouts.Select(layout => new SelectFieldItem(layout)));
			}
			return items;
		}

	}

}